api_end_end_testing/src/test/java/tds/student/api/tests/StudentApiTests.java

```diff
@@ -0,0 +1,726 @@
+package tds.student.api.tests;
+
+import static com.jayway.restassured.RestAssured.*;
+import static org.hamcrest.Matchers.*;
+
+import com.jayway.restassured.http.ContentType;
+import org.testng.annotations.Test;
+import tds.base.BaseUri;
+import tds.student.api.model.StudentDeleteBatchInfo;
+import tds.student.api.model.StudentInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+ * This class tests Student API endpoints that creates, updates and deletes a student
+ */
+public class StudentApiTests extends BaseUri {
+    private String uriLocation = "/rest/external/student/";
+
+    /*
+     * Test of Create Student, HTTP POST of /api/external/student/{stateCode}, 201 success item created
+     * Test of Get Student, HTTP GET of /api/external/student/{stateCode}/{ssid}, 200 success item found
+     */
+    private StudentInfo createStudent(String ssid, String stateAbbrev) {
+        StudentInfo studentInfo = new StudentInfo(
+            ssid,
+            stateAbbrev,
+            "DS9001",
+            "DISTRICT9",
+            "Amy",
+            "Martin",
+            "Leslie",
+            "1999-01-01",
+            "e" + ssid,
+            "04",
+            "Male",
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            true,
+            false,
+            false,
+            false,
+            false,
+            null,
+            false,
+            "2000-02-01",
+            null,
+            null,
+            null,
+            null,
+            0
+        );
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(studentInfo)
+        .when()
+            .post(uriLocation + stateAbbrev)
+        .then()
+            .statusCode(201)
+            .header("location", endsWith(uriLocation + stateAbbrev + "/" + ssid));
+
+        // Execute a GET by ssid to validate that the student was created
+        given()
+            .contentType(ContentType.JSON)
+            .accept(ContentType.JSON)
+            .header(authHeader)
+        .when()
+            .get(uriLocation + stateAbbrev + "/"  + ssid)
+        .then()
+            .statusCode(200)
+            .body("institutionIdentifier", is(studentInfo.getInstitutionIdentifier()))
+            .body("firstName", is(studentInfo.getFirstName()))
+            .body("middleName", is(studentInfo.getMiddleName()))
+            .body("lastName", is(studentInfo.getLastName()));
```

Comment:
Are these the only fields we can change via the API?  I guess I'm wondering if we need to verify all fields can be set/changed or just a subset.  I don't know the scope of these tests so I'm just curious not necessarily asking for a change.